docs/light/specification.md

```diff
@@ -1,356 +0,0 @@
-* Assign the current substore name to proof StoreName
-* Read multistore commitInfo from db by height and assign it to proof StoreCommitInfo
-
-Steps to verify proof:
-
-* If only right node exist, verify its exist proof and verify if it is the left most node
-* If only left node exist, verify its exist proof and verify if it is the right most node.
-* If both right node and left node exist, verify if they are adjacent.
-
-### Substores to AppHash Proof
-
-After verify the IAVL proof, then we can start to verify substore proof against AppHash. Firstly,
-iterate MultiStoreCommitInfo and find the substore commitID by proof StoreName. Verify if yhe Hash
-in commitID equals to proof RootHash. If not, the proof is invalid. Then sort the substore
-commitInfo array by the hash of substore name. Finally, build the simple Merkle tree with all
-substore commitInfo array and verify if the Merkle root hash equal to appHash.
-
-![substore proof](pics/substoreProof.png)
-
-```go
-func SimpleHashFromTwoHashes(left []byte, right []byte) []byte {
-    var hasher = ripemd160.New()
-
-    err := encodeByteSlice(hasher, left)
-    if err != nil {
-        panic(err)
-    }
-
-    err = encodeByteSlice(hasher, right)
-    if err != nil {
-        panic(err)
-    }
-
-    return hasher.Sum(nil)
-}
-
-func SimpleHashFromHashes(hashes [][]byte) []byte {
-    // Recursive impl.
-    switch len(hashes) {
-        case 0:
-            return nil
-        case 1:
-            return hashes[0]
-        default:
-            left := SimpleHashFromHashes(hashes[:(len(hashes)+1)/2])
-            right := SimpleHashFromHashes(hashes[(len(hashes)+1)/2:])
-            return SimpleHashFromTwoHashes(left, right)
-    }
-}
-```
-
-## Verify block header against validator set
-
-Above sections refer appHash frequently. But where does the trusted appHash come from? Actually,
-appHash exist in block header, so next we need to verify blocks header at specific height against
-LCD trusted validator set. The validation flow is shown as follows:
-
-![commit verification](pics/commitValidation.png)
-
-When the trusted validator set doesn't match the block header, we need to try to update our
-validator set to the height of this block. LCD have a rule that each validator set change should not
-affact more than 1/3 voting power. Compare with the trusted validator set, if the voting power of
-target validator set changes more than 1/3. We have to verify if there are hidden validator set
-change before the target validator set. Only when all validator set changes obey this rule, can our
-validator set update be accomplished.
-
-For instance:
-
-![Update validator set to height](pics/updateValidatorToHeight.png)
-
-* Update to 10000, tooMuchChangeErr
-* Update to 5050,  tooMuchChangeErr
-* Update to 2575, Success
-* Update to 5050, Success
-* Update to 10000,tooMuchChangeErr
-* Update to 7525, Success
-* Update to 10000, Success
-
-## Load Balancing
```

Comment:
I'd instead delete everything below this line. Not sure if the whole file

Reply:
Restored the top of the file